File: backend/FinTrack/Api.FunctionalTests/Contracts/CustomProblemDetails.cs
using SharedKernel;

namespace Api.FunctionalTests.Contracts;

internal sealed class CustomProblemDetails
{
    public required string Type { get; set; }

    public required string Title { get; set; }

    public required int Status { get; set; }

    public required string Detail { get; set; }

    public required List<Error> Errors { get; set; } = [];
}

File: frontend/fintrack_app/lib/core/errors/custom_problem_details.dart
import 'package:fintrack_app/core/errors/custom_error.dart';

final class CustomProblemDetails {
  final String type;
  final String title;
  final int status;
  final String detail;
  final List<CustomError> errors;

  const CustomProblemDetails({
    required this.type,
    required this.title,
    required this.status,
    required this.detail,
    required this.errors,
  });

  factory CustomProblemDetails.fromJson(Map<String, dynamic> json) {
    return CustomProblemDetails(
      type: json['type'] ?? '',
      title: json['title'] ?? '',
      status: json['status'] ?? 0,
      detail: json['detail'] ?? '',
      errors: (json['errors'] as List?)
              ?.map((e) => CustomError.fromJson(e))
              .toList() ??
          [],
    );
  }
}
